File: Java/heapsort/llama-3.2-90b-vision-preview/Llama_3_2_90b_vision_preview.java
import java.util.Random;
public class Llama_3_2_90b_vision_preview {

    /**
     * Sorts an array using the heapsort algorithm.
     *
     * @param array the array to be sorted
     */
    public static void sort(int[] array) {
        buildHeap(array);
        for (int end = array.length - 1; end > 0; end--) {
            swap(array, 0, end);
            heapify(array, 0, end);
        }
    }

    /**
     * Builds a max heap from the given array.
     *
     * @param array the array to build the heap from
     */
    private static void buildHeap(int[] array) {
        for (int i = array.length / 2 - 1; i >= 0; i--) {
            heapify(array, i, array.length);
        }
    }

    /**
     * Heapifies the given array starting from the specified index.
     *
     * @param array the array to heapify
     * @param index the index to start heapifying from
     * @param end the end index of the heap
     */
    private static void heapify(int[] array, int index, int end) {
        int largest = index;
        int left = 2 * index + 1;
        int right = 2 * index + 2;

        if (left < end && array[left] > array[largest]) {
            largest = left;
        }

        if (right < end && array[right] > array[largest]) {
            largest = right;
        }

        if (largest != index) {
            swap(array, index, largest);
            heapify(array, largest, end);
        }
    }

    /**
     * Swaps two elements in the given array.
     *
     * @param array the array to swap elements in
     * @param i the index of the first element
     * @param j the index of the second element
     */
    private static void swap(int[] array, int i, int j) {
        int temp = array[i];
        array[i] = array[j];
        array[j] = temp;
    }

    public static void main(String[] args) {
        Random random = new Random();

        // Tamanho do vetor
        int size = 1000;

        // Inicializando o vetor com 1000 números aleatórios entre 1 e 1000
        int[] array = new int[size];
        for (int i = 0; i < size; i++) {
            array[i] = random.nextInt(1000) + 1; // Gera números entre 1 e 1000
        }

        sort(array); // Ordena o array
    }

    /**
     * Prints the given array.
     *
     * @param array the array to print
     */
    private static void printArray(int[] array) {
        for (int element : array) {
            System.out.print(element + " ");
        }
        System.out.println();
    }
}

File: C/heapsort/llama-3.2-90b-vision-preview/llama-3.2-90b-vision-preview.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

// Function to swap two elements
void swap(int* a, int* b) {
    int temp = *a;
    *a = *b;
    *b = temp;
}

// Function to heapify the array
void heapify(int arr[], int n, int i) {
    int largest = i;
    int left = 2 * i + 1;
    int right = 2 * i + 2;

    if (left < n && arr[left] > arr[largest])
        largest = left;

    if (right < n && arr[right] > arr[largest])
        largest = right;

    if (largest != i) {
        swap(&arr[i], &arr[largest]);
        heapify(arr, n, largest);
    }
}

// Function to perform heap sort
void heapSort(int arr[], int n) {
    // Build a max heap
    for (int i = n / 2 - 1; i >= 0; i--)
        heapify(arr, n, i);

    // Extract elements one by one
    for (int i = n - 1; i >= 0; i--) {
        swap(&arr[0], &arr[i]);
        heapify(arr, i, 0);
    }
}

// Function to generate a random array
void generateRandomArray(int arr[], int n) {
    srand(time(NULL));
    for (int i = 0; i < n; i++)
        arr[i] = rand() % 1000;
}

int main() {
    int n = 1000;
    int* arr = (int*)malloc(n * sizeof(int));

    // Generate a random array
    generateRandomArray(arr, n);

    // Perform heap sort
    heapSort(arr, n);

    // Free the memory
    free(arr);

    return 0;
}